skcosmo/sample_selection/_voronoi_fps.py

```diff
@@ -0,0 +1,243 @@
+                voronoi_fps_timing = np.sum(voronoi_fps_times)
+                voronoi_fps_timing /= self.n_trial_calculation
+                if voronoi_fps_timing < simple_fps_timing:
+                    lower_fraction = self.full_fraction
+                else:
+                    top_fraction = self.full_fraction
+
+            self.full_fraction = lower_fraction
+        else:
+            if isinstance(self.full_fraction, numbers.Real):
+                if not 0 < self.full_fraction <= 1:
+                    raise ValueError(
+                        "Switching point should be real and more than 0 and less than 1",
+                        f"received {self.full_fraction}",
+                    )
+            else:
+                raise ValueError(
+                    "Switching point should be real and more than 0 and less than 1",
+                    f"received {self.full_fraction}",
+                )
+
+        super()._init_greedy_search(X, y, n_to_select)
+
+    def _continue_greedy_search(self, X, y, n_to_select):
+        """Continues the search. Prepares an array to store the selected
+        features."""
+
+        super()._continue_greedy_search(X, y, n_to_select)
+
+        n_pad = n_to_select - self.n_selected_
+
+        self.dSL_ = np.pad(self.dSL_, (0, n_pad), "constant", constant_values=0)
+
+    def _get_active(self, X, last_selected):
+        """
+        Finds the indices of the Voronoi cells that might change due to the fact
+        that a new point has been selected (index is last_selected,
+        because we haven't called super() yet, self.n_selected_ has not been
+        incremented yet)
+
+        This function calculates the distances between the last selected point
+        and the previously selected points. Next, the list of points to which
+        the distance is to be calculated is shortened.
+
+        These are the points for which need to recompute distances.
+        Let:
+        L is the last point selected;
+        S are the selected points from before this iteration;
+        X are the candidates;
+        The logic here is that we want to check if d(XL) can be smaller than
+        min(d(X,S)) (which is stored in self.haussdorf_)
+        now, if a point belongs to the Voronoi cell of S then
+        min(d(X,S_i))=d(X,S). Triangle inequality implies that
+        d(S,L) < |d(X,S) + d(L,X)| so we just need to check if
+        |d(X,S) - d(S,L)|>= d(X,S) to know that we don't need to check X.
+        but |d(X,S) - d(S,L)|^2>= d(X,S)^2 if and only if d(S,L)/2 > d(S,X)
+        """
+
+        if not hasattr(self, "n_selected_") or self.n_selected_ == 0:
+            return np.arange(X.shape[0], dtype=int)
+
+        else:
+            self.dSL_[: self.n_selected_] = (
+                self.norms_[self.selected_idx_[: self.n_selected_]]
+                + self.norms_[last_selected]
+                - 2 * (self.X_selected_[: self.n_selected_] @ X[last_selected].T)
+            ) * 0.25
```

Comment:
Can you explain to me why this is `* 0.25` and not `* 0.5`?

Reply:
Because here already squared distance. Initially we interested to compare d(LS)/2 and d(XS)<-----> compare d^2(LS)/4 and d^2(XS)